lib/sgov/sgov-adapter.ts

```diff
@@ -0,0 +1,227 @@
+      }
+    }
+
+    return sorted;
+  }
+
+  async getClass(cimIri: string): Promise<PimClass | null> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    const source = new SparqlQueryRdfSource(this.httpFetch, this.endpoint, getClassQuery(cimIri));
+    await source.query();
+    const entity = RdfSourceWrap.forIri(cimIri, source);
+
+    if (!await isSgovClass(entity)) {
+      return null;
+    }
+    await this.cacheResourceGroup(RdfSourceWrap.forIri(cimIri, source));
+    return await loadSgovClass(entity, this.iriProvider);
+  }
+
+  async getSurroundings(cimIri: string): Promise<ReadOnlyMemoryStore> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    const sources = getSurroundingsQueries(cimIri).map(query => new SparqlQueryRdfSource(this.httpFetch, this.endpoint, query));
+    await Promise.all(sources.map(q => q.query()));
+    const source = FederatedSource.createExhaustive(sources);
+    const resources = await this.loadPimEntitiesGraphFromEntity(cimIri, source);
+    return new ReadOnlyMemoryStore(resources);
+  }
+
+  public async getFullHierarchy(cimIri: string): Promise<ReadOnlyMemoryStore> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    const source = new SparqlQueryRdfSource(this.httpFetch, this.endpoint, getHierarchyQuery(cimIri));
+    await source.query();
+    const resources = await this.loadPimEntitiesGraphFromEntity(cimIri, source);
+    return new ReadOnlyMemoryStore(resources);
+  }
+
+  /**
+   * Returns a glossary info for the resource.
+   * @param cimIri
+   */
+  public async getResourceGroup(cimIri: string): Promise<string[]> {
+    if (!this.iriProvider) {
+      throw new Error("Missing IRI provider.");
+    }
+
+    if (this.resourceGroupCache.has(cimIri)) {
+      return this.resourceGroupCache.get(cimIri);
+    } else {
+      const source = new SparqlQueryRdfSource(this.httpFetch, this.endpoint, getGroupQuery(cimIri));
+      await source.query();
+      return await this.cacheResourceGroup(RdfSourceWrap.forIri(cimIri, source));
+    }
+  }
+
+  /**
+   * Takes an RdfSource and CIM iri of a class. From the class it loads everything accessible (associations, attributes
+   * and parents) and returns all the entities.
+   */
+  protected async loadPimEntitiesGraphFromEntity(rootClassCimIri: string, source: RdfSource): Promise<{ [iri: string]: CoreResource }> {
+    const resources: { [iri: string]: CoreResource } = {};
+
+    const classesProcessed = new Set<string>();
+    const associationsProcessed = new Set<string>();
+    // List of CIM iris to process
+    let processQueue: string[] = [rootClassCimIri];
+    while (processQueue.length) {
+      const processedCimIri = processQueue.pop();
+      if (classesProcessed.has(processedCimIri)) {
+        continue;
+      }
+      classesProcessed.add(processedCimIri);
+
+      // Class itself
+
```

Comment:
Why there is an empty line here, then there is none after the next comment and then again there is. It is not a big deal, but you should be consistent with your coding style.

Reply:
The reason is that if you place a comment (like this) right before the variable declaration, the IDE (JetBrains) thinks it belongs to the variable, not the "block".